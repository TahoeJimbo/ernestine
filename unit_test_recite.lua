
function recite_unittest()
   
   local saved_debug = DEBUG or nil

   playlist = {}

   playlist = recite.PRIV_make_phone_number_playlist(playlist, "9168011399")
   table_dump("9168011399", playlist)

   playlist = {}
   playlist = recite.PRIV_make_phone_number_playlist(playlist, "18314197859")
   table_dump("18314197859", playlist)

   playlist = {}
   playlist = recite.PRIV_make_phone_number_playlist(playlist, "19168011399")
   table_dump("19168011399", playlist)

   local playlist = {}
   playlist = recite.PRIV_make_phone_number_playlist(playlist, "4197859")
   table_dump("4197859", playlist)

   playlist = {}
   playlist = recite.PRIV_make_phone_number_playlist(playlist, "8011399")
   table_dump("8011399", playlist)

   playlist = {}
   playlsit = recite.PRIV_make_smart_digit_playlist(playlist, "9")
   table_dump("smart9", playlist)

   playlist = {}
   playlist = recite.PRIV_make_human_number_playlist(playlist, "9")
   table_dump("9", playlist)

   playlist = {}
   playlist = recite.PRIV_make_human_number_playlist(playlist, "43")
   table_dump("43", playlist)

   playlist = {}
   playlist = recite.PRIV_make_human_number_playlist(playlist, "123")
   table_dump("123", playlist)

   playlist = {}
   playlist = recite.PRIV_make_human_number_playlist(playlist, "8019")
   table_dump("8019", playlist)
   
   DEBUG = saved_debug
end
