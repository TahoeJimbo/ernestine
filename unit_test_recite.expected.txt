9168011399
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/area-code.wav
2	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/9-beg.wav
3	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/1-mid-short.wav
4	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/6-mid-long.wav
5	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/8-beg.wav
6	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/0-mid-short.wav
7	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/1-mid-long.wav
8	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/1-beg.wav
9	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/3-mid-long.wav
10	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/9-mid-short.wav
11	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/9-end.wav
18314197859
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/ac-831.wav
2	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/419-beg.wav
3	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/78-beg.wav
4	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/59-end.wav
19168011399
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/area-code.wav
2	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/9-beg.wav
3	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/1-mid-short.wav
4	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/6-mid-long.wav
5	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/8-beg.wav
6	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/0-mid-short.wav
7	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/1-mid-long.wav
8	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/1-beg.wav
9	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/3-mid-long.wav
10	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/9-mid-short.wav
11	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/9-end.wav
4197859
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/419-beg.wav
2	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/78-beg.wav
3	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/59-end.wav
8011399
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/8-beg.wav
2	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/0-mid-short.wav
3	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/1-mid-long.wav
4	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/1-beg.wav
5	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/3-mid-long.wav
6	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/9-mid-short.wav
7	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/9-end.wav
smart9
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/phone/9-end.wav
9
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/9.wav
43
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/40.wav
2	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/3.wav
123
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/1.wav
2	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/hundred.wav
3	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/20.wav
4	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/3.wav
8019
1	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/8.wav
2	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/thousand.wav
3	/usr/local/freeswitch/sounds/en/us/kevin/Numbers/cardinal/19.wav
